File: javascript/sort.js
function shuffle(arra1) {
    let ctr = arra1.length;
    let temp;
    let index;
   while (ctr > 0) {
        index = Math.floor(Math.random() * ctr);
        ctr--;
        temp = arra1[ctr];
        arra1[ctr] = arra1[index];
        arra1[index] = temp;
    }
    return arra1;
}


console.log("Ordenador de listas - JAVASCRIPT");

let inicio = new Date().getTime();
const potencia = process.argv[2] || 1;
const total = 10 ** potencia;
console.log("ordena lista com tamanho "+total +" (10^" + potencia + ")");
console.log("criando lista...");
let lista = [...Array(total).keys()].map(i => i + 1);
console.log("embaralhando...");
let embaralhada = shuffle(lista);
console.log("ordenando...");
let ordenada = embaralhada.sort((a, b) => a - b);
let fim = new Date().getTime();
let tempo = (fim - inicio) / 1000; // em ms

console.log(tempo + " s");

File: python/sort.py
from random import shuffle
import datetime

def main():
    print("Ordenador de listas - PYTHON")
    potencia = int(input("Digite a quantidade de elementos(10 ^ x):"))
    total  = 10 ** potencia
    start_time = datetime.datetime.now()
    lista = range(total)
    print("ordena lista com tamanho {} (10^{})".format(len(lista), potencia))
    # print('lista original: {}'.format(lista))
    print("embaralhando...")
    shuffle(lista)
    # print('lista embaralhada: {}'.format(lista))
    print("ordenando...")
    lista.sort()
    elapsed_time = datetime.datetime.now() - start_time
    # print('lista ordenada: {}'.format(lista))
    print('tempo total: {}:{}'.format(elapsed_time.seconds, elapsed_time.microseconds))

main()

